news/_posts/2022-07-01-introducing-horizontal-migration.md

```diff
@@ -0,0 +1,74 @@
+
+Migrating a page this way is slow and requires a considerable effort. It basically amounts to building the page again – a _better_ version of it. With several dozen modified files, refactoring a typical page takes several weeks of development, and months of review and refinement. There are still 50 pages left to migrate in the Back office. That's _years'_ worth of work! 
+
+In the meantime, PrestaShop's architecture has become heterogeneous. Although the pages migrated to Symfony benefit from all the modern concepts built over the years, a third of the Back office pages remain largely identical to how they were in PrestaShop 1.6 – not to mention the whole Front office and the Webservice API. 
+
+This means that developers have to deal with some parts of the software working in a certain way and others in a completely different one. As we have mentioned in previous articles, this is bad because having more than one way of doing the same thing carries a higher risk of bugs, lower performance, more maintenance work, and a steeper learning curve for people who are new to the platform.
+
+Moreover, every bit of the legacy framework must be preserved until the migration has been fully completed. Even if in a few years only a single page that few people use was left to migrate, everything it relies on, Dispatcher, ObjectModel, Smarty, Helpers... the entire legacy stack must remain in place, alive and maintained.
+
+Of course, in time, the migration will be finished, we will be able to remove any remaining legacy components, and reduce the platform's complexity. But as the saying goes, "in the long-term, we will all be dead". PrestaShop's attractiveness as a platform is at risk if we don't manage to accelerate the migration.
+
+**We need a better strategy.** How can we accelerate the transition to Symfony, increase the platform's homogeneity, and phase out legacy components faster?
+
+## Thinking outside the box
+
+Let's analyze our current strategy. Migrating a page takes a list of known steps. Currently, we perform all those steps on a single page, then move on to the next page, and so on. 
+
+What if we focused on delivering a single step at a time, progressively across all pages? 
+
+Take the controller part, for example. At its root, a Symfony controller takes a Request and provides a Response, that's all. How the request is processed in order to build that response has no impact on the nature of the controller. Who says Symfony controller must use Twig? Or Grid? Or Symfony forms, even? Theoretically, it should be possible to take the content of the legacy controller and paste it into a Symfony controller. As long as the original code is rearranged into actions, and we get a response, it doesn't matter if the output is built using Smarty helpers: HTML is HTML.
```

Comment:
The capitals are used because I'm referring to Symfony components that are called Request and Response.

Alternative:
```suggestion
Take the controller part, for example. At its root, a Symfony controller takes a Request and provides a Response, that's all. How the request is processed in order to build that response has no impact on the nature of the controller. Who says a Symfony controller must use Twig? Or Grid? Or Symfony forms, for that matter? Theoretically, it should be possible to take the content of the legacy controller and paste it into a Symfony controller. As long as the original code is rearranged into actions, and we get a response, it doesn't matter if the output is built using Smarty helpers: HTML is HTML.
```

Reply:
Before making the edit, I checked on the Symfony documentation and it seems they don't systematically use capital letters for both request and response, but happy to keep the capitals on the article if you feel it's needed: https://symfony.com/doc/current/introduction/http_fundamentals.html

Reply:
Yep but in that part they are referring to the generic term. This part for example refers to the actual components: https://symfony.com/doc/4.4/controller.html#the-request-object-as-a-controller-argument

We could add code fences to it to clarify if you think it makes sense
```suggestion
Take the controller part, for example. At its root, a Symfony controller takes a `Request` and provides a `Response`, that's all. How the request is processed in order to build that response has no impact on the nature of the controller. Who says a Symfony controller must use Twig? Or Grid? Or Symfony forms, for that matter? Theoretically, it should be possible to take the content of the legacy controller and paste it into a Symfony controller. As long as the original code is rearranged into actions, and we get a response, it doesn't matter if the output is built using Smarty helpers: HTML is HTML.
```